packages/@aws-cdk/core/lib/asset-staging.ts

```diff
@@ -281,11 +300,34 @@ export class AssetStaging extends CoreConstruct {
     const bundleDir = this.determineBundleDir(this.assetOutdir, assetHash);
     this.bundle(bundling, bundleDir);
 
+    let assetPath = bundleDir;
+    let extension: string | undefined;
+
+    const packaging = bundling.packaging ?? BundlePackaging.AUTO;
+    if (packaging !== BundlePackaging.ALWAYS_ZIP) {
```

Comment:
Please extract to a helper method and consider using a switch statement to improve readability. The logic is slightly too convoluted.